micov/cli.py

```diff
@@ -273,5 +274,38 @@ def _load_db(dbbase, sample_metadata, features_to_keep):
                                               FROM coverage)""")
 
 
+@cli.command()
+@click.option('-pos', '--covered-positions', type=click.Path(exists=True),
+              required=True,
+              help='Covered positions calculated from one or more samples')
+@click.option('-o', '--outdir', type=click.Path(exists=True),
+              required=True, help="Output directory. If new, will be created.")
+@click.option('-g', '--genome-id', type=str,
+              required=True, help="Genome ID of the genome of interest")
+@click.option('-l', '--genome-length', type=int,
+              required=True, help="Length of the genome of interest")
+@click.option('-n', '--bin-num', type=int, default=1000,
+              required=False, help="Number of bins")
+def binning(covered_positions, outdir, genome_id, genome_length, bin_num):
+    """Bin genome positions and quantify read and sample hits across bins."""
+    pos = pl.read_csv(covered_positions, separator='\t',
+                      new_columns=BED_COV_SAMPLEID_SCHEMA.columns,
+                      schema_overrides=BED_COV_SAMPLEID_SCHEMA.dtypes_dict,
+                      has_header=False, skip_rows=1).lazy()
```

Comment:
Shouldn't this be `has_header=True` and no skipped rows?

I think in other parts of micov, we dump the covered positions as parquet? If so then supporting that here would be nice

Reply:
yes, need to support parquet. Propose doing that in a separate PR. 

`schema_overrides=BED_COV_SAMPLEID_SCHEMA.dtypes_dict` errored out when changing `has_header=True` with no skipped rows. 
```
pos = pl.read_csv("./input_test/BE_All.covered_positions", separator='\t',
                      new_columns=BED_COV_SAMPLEID_SCHEMA.columns,
                      schema_overrides=BED_COV_SAMPLEID_SCHEMA.dtypes_dict,
                      has_header=True)

TypeError: argument 'overwrite_dtype_slice': 'type' is not a Polars data type
```
This error message is confusing... Maybe they meant {type} instead of 'type'? Anyways I investigated more on this error. For some reason, whenever `has_header=True`, `BED_COV_SAMPLEID_SCHEMA.dtypes_dict` needs to specify [polar-specific data types](https://docs.pola.rs/api/python/stable/reference/datatypes.html) like `pl.String` or `pl.Int64` instead of generic data types like `str` or `int`. So when
```
 BED_COV_SAMPLEID_SCHEMA.dtypes_dict = {
    'genome_id': pl.String,
    'start': pl.Int64,
    'stop': pl.Int64,
    'sample_id': pl.String
}
```
```
pos = pl.read_csv("./input_test/BE_All.covered_positions", separator='\t',
                      new_columns=BED_COV_SAMPLEID_SCHEMA.columns,
                      schema_overrides=BED_COV_SAMPLEID_SCHEMA.dtypes_dict,
                      has_header=True)
```
passed with no error. 

To deal with this strange behavior, I suggest keeping `has_header=False, skip_rows=1` for now. Alternatively we can change the value of `BED_COV_SAMPLEID_SCHEMA.dtypes_dict` and also update other read_csv functions previously written (for example: [here](https://github.com/biocore/micov/blob/main/micov/_io.py#L42)).

Reply:
Okay, thanks for tracking, agree with proposal, and it will get cleaner with centralized schemas